akka-cluster/src/test/scala/akka/cluster/GossipSpec.scala

```diff
@@ -340,15 +340,18 @@ class GossipSpec extends WordSpec with Matchers {
 
     "clear out a bunch of stuff when removing a node" in {
       val g = Gossip(members = SortedSet(dc1a1, dc1b1, dc2d2))
+        .:+(VectorClock.Node(Gossip.vclockName(dc1b1.uniqueAddress)))
+        .:+(VectorClock.Node(Gossip.vclockName(dc2d2.uniqueAddress)))
         .remove(dc1b1.uniqueAddress, System.currentTimeMillis())
 
       g.seenBy should not contain (dc1b1.uniqueAddress)
       g.overview.reachability.records.exists(_.observer == dc1b1.uniqueAddress) should be(false)
       g.overview.reachability.records.exists(_.subject == dc1b1.uniqueAddress) should be(false)
-      g.version.versions should have size (0)
 
       // sort order should be kept
       g.members.toList should ===(List(dc1a1, dc2d2))
+      g.version.versions.keySet should not contain (VectorClock.Node(Gossip.vclockName(dc1b1.uniqueAddress)))
+      g.version.versions.keySet should contain(VectorClock.Node(Gossip.vclockName(dc2d2.uniqueAddress)))
```

Comment:
Perhaps add a check above that there *are* reachability records for `dc2d2` for completeness, though this is OK.